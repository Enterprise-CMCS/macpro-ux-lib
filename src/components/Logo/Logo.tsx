import React from "react";
import { Typography } from "../Typography/Typography";

type IntrinsicElements = JSX.IntrinsicElements["img"];

export interface LogoProps extends IntrinsicElements {
  altText: string;
  ariaHidden?: boolean;
  ariaLabel?: string;
  className?: string;
  height?: number;
  role?: string;
  source?: string;
  width?: number;
}

/**
 * Logo Component
 * @param {string} altText        Alternate image text.
 * @param {string} ariaLabel      Describes the image for a screen reader.
 * @param {string} [ariaHidden]   Determines whether or not the element is hidden from a screen reader.
 * @param {string} [className]    Optional additional className for the logo container.
 * @param {string} [height]       Height parameter for logo in # of pixels.
 * @param {string} [role]         Describes the role of an element in programs that can make use of it.
 * @param {string} [source]       Source of the logo image. This can be an image file or a web address pointing to an image.
 * @param {string} [width]        Width parameter for logo in # of pixels.
 */

export const Logo: React.FC<LogoProps> = ({
  altText,
  ariaHidden = false,
  ariaLabel,
  className,
  height,
  role = "img",
  source,
  width,
  ...rest
}) => {
  return (
    <div
      role={role}
      aria-label={ariaLabel || altText}
      className={`application-logo-container ${className ?? ""}`}
    >
      {!source && (
        <Typography as="h2" className="usa-logo__text">
          {altText}
        </Typography>
      )}
      {source && (
        <img
          alt={altText}
          aria-hidden={ariaHidden}
          height={height}
          src={source}
          width={width}
          {...rest}
        />
      )}
    </div>
  );
};
