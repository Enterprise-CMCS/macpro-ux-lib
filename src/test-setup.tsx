import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
