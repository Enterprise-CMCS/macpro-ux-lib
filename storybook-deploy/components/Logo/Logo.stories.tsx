import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Logo } from "./Logo";
import pngLogo from "../../assets/img/logos/cms_logo.png";
import svgLogo from "../../assets/img/logos/cms_logo.svg";

export default {
  title: "USWDS/Base/Logo",
  component: Logo,
  args: {
    altText: "CMS.gov Logo",
    ariaLabel: "CMS.gov Logo",
    ariaHidden: false,
    role: "img",
    width: 200,
  },
  argTypes: {
    altText: {
      description: "Alternate image text.",
    },
    ariaHidden: {
      description:
        "Determines whether or not the element is hidden from a screen reader.",
    },
    ariaLabel: {
      description: "Describes the image for a screen reader.",
    },
    className: {
      description: "Optional additional className for the logo container.",
    },
    height: {
      description: "Height parameter for logo in # of pixels.",
    },
    role: {
      description:
        "Describes the role of an element in programs that can make use of it.",
    },
    source: {
      description:
        "Source of the logo image. This can be an image file or a web address pointing to an image.",
    },
    width: {
      description: "Width parameter for logo in # of pixels.",
    },
  },
} as ComponentMeta<typeof Logo>;

const Template: ComponentStory<typeof Logo> = ({ ...rest }) => (
  <Logo {...rest} />
);

export const SVGLogo = Template.bind({});
export const PNGLogo = Template.bind({});
export const ExternalSourceLogo = Template.bind({});
SVGLogo.args = {
  source: svgLogo,
};

PNGLogo.args = {
  source: pngLogo,
};

ExternalSourceLogo.args = {
  source: "https://www.cms.gov/themes/custom/cms_evo/logo.svg",
  altText:
    "This logo is hosted on CMS.gov. If you're reading this the image has likely been moved.",
};
