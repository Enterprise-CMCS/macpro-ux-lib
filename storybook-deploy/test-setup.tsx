import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };

/**
 * Clean up attributes from the rendered component by removing the generated ids.
 *
 * Before:
 *   for="input-type-text-123456"
 *   id="input-type-text-123456"
 *
 * After:
 *   for="input-type-text"
 *   id="input-type-text"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 * @param {prefixes}      string[]  The reference substrings to leave alone.
 */
