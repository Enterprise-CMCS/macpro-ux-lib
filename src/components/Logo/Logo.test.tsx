import React from "react";
import { render } from "../../test-setup";

import { Logo } from "./Logo";
import pngLogo from "../../assets/img/logos/cms_logo.png";
import svgLogo from "../../assets/img/logos/cms_logo.svg";

describe("Tests for the button component.", () => {
  it("Should render an svg", () => {
    const { container } = render(
      <Logo altText="CMS.gov Logo" ariaLabel="CMS.gov Logo" source={svgLogo} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Should render a png and have unique className", () => {
    const { container } = render(
      <Logo
        altText="CMS.gov Logo"
        ariaLabel="CMS.gov Logo"
        source={pngLogo}
        className={"beep-boop"}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
