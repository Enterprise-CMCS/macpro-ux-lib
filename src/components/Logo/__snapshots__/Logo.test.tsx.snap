// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Logo component Should render a png and have unique className 1`] = `
<div>
  <div
    aria-label="CMS.gov Logo"
    class="application-logo-container beep-boop"
    role="img"
  >
    <img
      alt="CMS.gov Logo"
      aria-hidden="false"
      src="[object Object]"
      width="200"
    />
  </div>
</div>
`;

exports[`Tests for the Logo component Should render an svg 1`] = `
<div>
  <div
    aria-label="CMS.gov Logo"
    class="application-logo-container "
    role="img"
  >
    <img
      alt="CMS.gov Logo"
      aria-hidden="false"
      src="[object Object]"
      width="200"
    />
  </div>
</div>
`;
